examples/server/server.cpp

```diff
@@ -2200,7 +2241,7 @@ struct server_context {
                 pos = std::min(slot.n_sent_text, slot.generated_text.size());
             } else if (slot.has_next_token) {
                 stop_pos = slot.find_stopping_strings(str_test, token_str.size(), false);
-                send_text = stop_pos == std::string::npos;
+                send_text = send_text && stop_pos == std::string::npos;
             }
 
             // check if there is any token to predict
```

Comment:
Upon closer inspection, I think manipulating `send_text` is not an optimum solution here.

If you look at the streamed response, when start string is not yet found, the server will still send an empty string for each token generated. See the `else` branch of `if (send_text)` below.

Instead, I think this logic should be placed outside this `if (!incomplete)`